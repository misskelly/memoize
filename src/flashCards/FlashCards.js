import React from 'react';
import Card from '../card/Card'

export default function FlashCards(props) {
  const { getRandomCard, removeFromDeck, deck, card, updateDeck } = props;
  return (
    <section className='flashCards'>
      {
        !deck.length &&
        <article className='noCardsContainer'>
          <h5 className='noCardsMessage'>You made it through the entire deck! Want to review some more?</h5>
          <button type='submit'
                  className='refreshBtn'>
            Mo' Memoize Mo' Better
          </button>
        </article>
      }
      {
        deck.length !== 0 &&
        <Card cardId={card.cardNumber}
        term={card.term}
        deck={deck}
        definition={card.definition}
        getRandomCard={getRandomCard}
        removeFromDeck={removeFromDeck}
        updateDeck={updateDeck}
        
      // addDuplicate={props.card.addDuplicate} 
      />
    }
          
    </section>
  )
}