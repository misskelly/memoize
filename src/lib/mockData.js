const mockCards = [
  {
    "categoryId": 1,
    "cardNumber": 101,
    "term": "React",
    "definition": "A client-side JavaScript framework that allows you to easily and efficiently manipulate the DOM based on application data and how it changes in response to user interaction"
  },
  {
    "categoryId": 1,
    "cardNumber": 102,
    "term": "The Virtual DOM",
    "definition": "An in-memory object that represents a DOM structure and can be manipulated with JavaScript before updating the real DOM"
  },
  {
    "categoryId": 1,
    "cardNumber": 103,
    "term": "JSX",
    "definition": "A mix of JavaScript and XML that facilitates rendering the appropriate HTML, lives inside a component"
  },
  {
    "categoryId": 2,
    "cardNumber": 205,
    "term": "Enzyme",
    "definition": "A JavaScript Testing utility for React that makes it easier to test your React Components' output.  It allows test files to be more readable and provides us with helper methods for testing common pieces of functionality."
  },
]

const mockCurrentCard = {
  "categoryId": 1,
  "cardNumber": 103,
  "term": ".simulate()",
  "definition": "Simulates an event on an element by passing the event name string in as an argument."
}


module.exports = {
  mockCards,
  mockCurrentCard
}