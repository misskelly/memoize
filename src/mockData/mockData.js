const mockCards = [
  {
    "categoryId": 1,
    "cardNumber": 101,
    "term": "React",
    "definition": "A client-side JavaScript framework that allows you to easily and efficiently manipulate the DOM based on application data and how it changes in response to user interaction"
  },
  {
    "categoryId": 1,
    "cardNumber": 102,
    "term": "The Virtual DOM",
    "definition": "An in-memory object that represents a DOM structure and can be manipulated with JavaScript before updating the real DOM"
  },
  {
    "categoryId": 1,
    "cardNumber": 103,
    "term": "JSX",
    "definition": "A mix of JavaScript and XML that facilitates rendering the appropriate HTML, lives inside a component"
  },
  {
    "categoryId": 1,
    "cardNumber": 104,
    "term": "Component",
    "definition": "Reusable pieces of code that represent templates for a particular instance of a UI element. They are either functions or an extended ES6 class"
  }
]

const mockCurrentCard = {
  "categoryId": 2,
  "cardNumber": 214,
  "term": ".simulate()",
  "definition": "Simulates an event on an element by passing the event name string in as an argument."
}


module.exports = {
  mockCards,
  mockCurrentCard
}