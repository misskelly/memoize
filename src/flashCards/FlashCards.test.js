import React from 'react';
import FlashCards from './FlashCards';
import { mockCards, mockCurrentCard } from '../mockData/mockData'
import { shallow } from 'enzyme';

const mockRemoveFromDeck = jest.fn();
const mockGetRandomCard = jest.fn();

describe('FlashCards', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <FlashCards deck={mockCards}
                  card={mockCurrentCard}
                  getRandomCard={mockGetRandomCard}
                  removeFromDeck={mockRemoveFromDeck}/>
    )
  });

  it('should match snapshot', () => {

    expect(wrapper).toMatchSnapshot();
  });

});