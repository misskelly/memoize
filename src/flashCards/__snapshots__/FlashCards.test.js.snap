// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlashCards should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FlashCards
    card={
      Object {
        "cardNumber": 214,
        "categoryId": 2,
        "definition": "Simulates an event on an element by passing the event name string in as an argument.",
        "term": ".simulate()",
      }
    }
    deck={
      Array [
        Object {
          "cardNumber": 101,
          "categoryId": 1,
          "definition": "A client-side JavaScript framework that allows you to easily and efficiently manipulate the DOM based on application data and how it changes in response to user interaction",
          "term": "React",
        },
        Object {
          "cardNumber": 102,
          "categoryId": 1,
          "definition": "An in-memory object that represents a DOM structure and can be manipulated with JavaScript before updating the real DOM",
          "term": "The Virtual DOM",
        },
        Object {
          "cardNumber": 103,
          "categoryId": 1,
          "definition": "A mix of JavaScript and XML that facilitates rendering the appropriate HTML, lives inside a component",
          "term": "JSX",
        },
        Object {
          "cardNumber": 104,
          "categoryId": 1,
          "definition": "Reusable pieces of code that represent templates for a particular instance of a UI element. They are either functions or an extended ES6 class",
          "term": "Component",
        },
      ]
    }
    getRandomCard={[MockFunction]}
    removeFromDeck={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Card
        cardId={214}
        definition="Simulates an event on an element by passing the event name string in as an argument."
        getRandomCard={[MockFunction]}
        removeFromDeck={[MockFunction]}
        term=".simulate()"
      />,
      "className": "flashCards",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "cardId": 214,
        "definition": "Simulates an event on an element by passing the event name string in as an argument.",
        "getRandomCard": [MockFunction],
        "removeFromDeck": [MockFunction],
        "term": ".simulate()",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Card
          cardId={214}
          definition="Simulates an event on an element by passing the event name string in as an argument."
          getRandomCard={[MockFunction]}
          removeFromDeck={[MockFunction]}
          term=".simulate()"
        />,
        "className": "flashCards",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "cardId": 214,
          "definition": "Simulates an event on an element by passing the event name string in as an argument.",
          "getRandomCard": [MockFunction],
          "removeFromDeck": [MockFunction],
          "term": ".simulate()",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
